tensornetwork/matrixproductstates/mps.py

```diff
@@ -139,9 +427,12 @@ def position(self, site: int, normalize: Optional[bool] = True) -> np.number:
             left_edges=[self.nodes[n][0], self.nodes[n][1]],
             right_edges=[self.nodes[n][2]],
             left_name=self.nodes[n].name)
-
+        Q[2] | R[0]  #break the edge between Q and R
+        order = [R[0], self.nodes[n + 1][1], self.nodes[n + 1][2]]
+        R[1] ^ self.nodes[n + 1][0]  #connect R to the right node
         self.nodes[n] = Q  #Q is a left-isometric tensor of rank 3
         self.nodes[n + 1] = contract(R[1], name=self.nodes[n + 1].name)
+        self.nodes[n + 1].reorder_edges(order)
```

Comment:
I'd rather like to stick to numbers, in order to  keep the tensors in a well defined order (i.e. (D,d,D)).
Not sure, but if we switch to names the order of the underlying tensors seems to be random 
(unless we again order them explicitly). This would force us to use names everywhere, and could be potentially confusing

Reply:
Yes, you would have to use names everywhere, but I see that as a feature, rather than a bug :)

Reply:
It could be, but it might be potentially annoying as well, in particular if we have to frequently access the underlying tensors, which happens when passing them around e.g. sparse eigensolvers in the backends. Another question is naming:  should names be unique? if no then we might run into the uniquifaction problem again, if yes that's potentially anoying to code (adding labels to strings)....

Reply:
Well, it's easy to enforce an ordering before passing tensors to other code, but the labels would be kind of pointless if that had to happen a lot, I agree. Most of the code in mps.py looks like it wouldn't need explicit ordering, and I don't see any obvious uniqueness problems. It's hard to say without giving it a try though... perhaps I will do that at some point!

Reply:
I'd say we keep it as is for now but keep on thinking about it. I'd very much like to get rid of
index conventions, but I'd like to first have the thing running and change it later on.

Reply:
👍